src/client/components/filter-tile/filter-tiles-row.tsx

```diff
@@ -0,0 +1,211 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as React from "react";
+import { Dimension } from "../../../common/models/dimension/dimension";
+import { DragPosition } from "../../../common/models/drag-position/drag-position";
+import { FilterClause } from "../../../common/models/filter-clause/filter-clause";
+import { Locale } from "../../../common/models/locale/locale";
+import { Stage } from "../../../common/models/stage/stage";
+import { Timekeeper } from "../../../common/models/timekeeper/timekeeper";
+import { Binary } from "../../../common/utils/functional/functional";
+import { Fn } from "../../../common/utils/general/general";
+import { CORE_ITEM_GAP, CORE_ITEM_WIDTH, STRINGS } from "../../config/constants";
+import { getXFromEvent, setDragData, setDragGhost } from "../../utils/dom/dom";
+import { DragManager } from "../../utils/drag-manager/drag-manager";
+import { getMaxItems } from "../../utils/pill-tile/pill-tile";
+import { CubeContext, CubeContextValue } from "../../views/cube-view/cube-context";
+import { PartialFilter } from "../../views/cube-view/partial-tiles-provider";
+import { DragIndicator } from "../drag-indicator/drag-indicator";
+import { AddFilter } from "./add-filter";
+import "./filter-tile.scss";
+import { FilterTiles } from "./filter-tiles";
+
+interface FilterTilesRowProps {
+  menuStage: Stage;
+  timekeeper: Timekeeper;
+  locale: Locale;
+  removePartialFilter: Fn;
+  addPartialFilter: Binary<Dimension, DragPosition, void>;
+  partialFilter?: PartialFilter;
+}
+
+interface FilterTilesRowState {
+  dragPosition?: DragPosition;
+  openedClause?: FilterClause;
+  overflowOpen?: boolean;
+}
+
+export class FilterTilesRow extends React.Component<FilterTilesRowProps, FilterTilesRowState> {
+  static contextType = CubeContext;
+  context: CubeContextValue;
+
+  state: FilterTilesRowState = {};
+  private items = React.createRef<HTMLDivElement>();
+
+  private maxItems(): number {
+    const { essence: { filter } } = this.context;
+    const { menuStage } = this.props;
+    return menuStage && getMaxItems(menuStage.width, filter.length());
+  }
+
+  openFilterMenu = (clause: FilterClause) => this.setState({ openedClause: clause });
+
+  closeFilterMenu = () => this.setState({ openedClause: null });
+
+  openOverflowMenu = () => this.setState({ overflowOpen: true });
+
+  closeOverflowMenu = () => this.setState({ overflowOpen: false });
+
+  canDrop(): boolean {
```

Comment:
A lot of the code here is very similar to SeriesTilesRow and SplitTilesRow but they differ even so slightly. I don't think we should push for DRY here.

Reply:
If you suspect, that in the future this code would drift apart from SeriesTilesRow and SplitTilesRow I would recommend keeping it separate. However, if that's not the case, I would merge it with the rest